tests/property/test_decryption_mediator.py

```diff
@@ -0,0 +1,327 @@
+                self.internal_manifest.ballot_styles[0].object_id
+            )
+        )
+        self.assertTrue(
+            self.fake_spoiled_ballot.is_valid(
+                self.internal_manifest.ballot_styles[0].object_id
+            )
+        )
+        self.expected_plaintext_tally = accumulate_plaintext_ballots(
+            [self.fake_cast_ballot] + more_fake_ballots
+        )
+
+        # Fill in the expected values with any missing selections
+        # that were not made on any ballots
+        selection_ids = {
+            selection.object_id
+            for contest in self.internal_manifest.contests
+            for selection in contest.ballot_selections
+        }
+
+        missing_selection_ids = selection_ids.difference(
+            set(self.expected_plaintext_tally)
+        )
+
+        for id in missing_selection_ids:
+            self.expected_plaintext_tally[id] = 0
+
+        # Encrypt
+        self.encrypted_fake_cast_ballot = self.ballot_marking_device.encrypt(
+            self.fake_cast_ballot
+        )
+        self.encrypted_fake_spoiled_ballot = self.ballot_marking_device.encrypt(
+            self.fake_spoiled_ballot
+        )
+        self.assertIsNotNone(self.encrypted_fake_cast_ballot)
+        self.assertIsNotNone(self.encrypted_fake_spoiled_ballot)
+        self.assertTrue(
+            self.encrypted_fake_cast_ballot.is_valid_encryption(
+                self.internal_manifest.manifest_hash,
+                self.joint_public_key.joint_public_key,
+                self.context.crypto_extended_base_hash,
+            )
+        )
+
+        # encrypt some more fake ballots
+        more_fake_encrypted_ballots = []
+        for fake_ballot in more_fake_ballots:
+            more_fake_encrypted_ballots.append(
+                self.ballot_marking_device.encrypt(fake_ballot)
+            )
+        # encrypt some more fake ballots
+        self.more_fake_encrypted_spoiled_ballots = []
+        for fake_ballot in more_fake_spoiled_ballots:
+            self.more_fake_encrypted_spoiled_ballots.append(
+                self.ballot_marking_device.encrypt(fake_ballot)
+            )
+
+        # configure the ballot box
+        ballot_store = DataStore()
+        ballot_box = BallotBox(self.internal_manifest, self.context, ballot_store)
+        ballot_box.cast(self.encrypted_fake_cast_ballot)
+        ballot_box.spoil(self.encrypted_fake_spoiled_ballot)
+
+        # Cast some more fake ballots
+        for fake_ballot in more_fake_encrypted_ballots:
+            ballot_box.cast(fake_ballot)
+        # Spoil some more fake ballots
+        for fake_ballot in self.more_fake_encrypted_spoiled_ballots:
+            ballot_box.spoil(fake_ballot)
+
+        # generate encrypted tally
+        self.ciphertext_tally = tally_ballots(
+            ballot_store, self.internal_manifest, self.context
+        )
+        self.ciphertext_ballots = get_ballots(ballot_store, BallotBoxState.SPOILED)
```

Comment:
lol @ the length of this method :-p honestly though, i dont think you should change it because this is a unit test and we are trying to show all of the steps of a linear process.  breaking this out into small functions will decrease readability in my opinion

Reply:
I would like to venture a suggestion that we remove this a different way. This should be handled in `Helper`, `Factory` or `Builder` in the `electionguardtest`/`electionguard-mock` setup. 

`KeyCeremonyHelper.perform_full_ceremony`
`BallotBoxHelper.mock_voting`
`TallyHelper.tally_ballots`

Reply:
agreed